store/store.go

```diff
@@ -428,7 +429,17 @@ func (s *SnapUbuntuStoreRepository) Snap(name, channel string, auther Authentica
 
 	s.checkStoreResponse(resp)
 
-	return infoFromRemote(searchData.Payload.Packages[0]), nil
+	pkg := searchData.Payload.Packages[0]
+
+	var purchases []*purchase
+	if auther != nil {
+		purchases, err = s.getPurchases(pkg.SnapID, channel, auther)
```

Comment:
and even if we must/want to do this query always, we should avoid it if we know the snap is free